e2e/test/scenarios/embedding/sdk-iframe-embedding/authentication.cy.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { ORDERS_DASHBOARD_ID } from "e2e/support/cypress_sample_instance_data";
+import {
+  mockAuthSsoEndpointForSamlAuthProvider,
+  stubWindowOpenForSamlPopup,
+} from "e2e/support/helpers/embedding-sdk-testing";
+
+const { H } = cy;
+
+describe("scenarios > embedding > sdk iframe embedding > authentication", () => {
+  it("cannot login if no auth methods are enabled", () => {
+    H.prepareSdkIframeEmbedTest({ enabledAuthMethods: [] });
+    cy.signOut();
+
+    const frame = H.loadSdkIframeEmbedTestPage({
+      dashboardId: ORDERS_DASHBOARD_ID,
+    });
+
+    frame.within(() => {
+      cy.findByTestId("sdk-error-container")
+        .should("be.visible")
+        .and("contain", "Backend returned an error when refreshing the token.");
+    });
+  });
+
+  it("can login via JWT", () => {
+    H.prepareSdkIframeEmbedTest({ enabledAuthMethods: ["jwt"] });
+    cy.signOut();
+
+    const frame = H.loadSdkIframeEmbedTestPage({
+      dashboardId: ORDERS_DASHBOARD_ID,
+    });
+
+    assertDashboardLoaded(frame);
+  });
+
+  it("can login via SAML", () => {
+    mockAuthSsoEndpointForSamlAuthProvider();
+    H.prepareSdkIframeEmbedTest({ enabledAuthMethods: [] });
+    cy.signOut();
+
+    const frame = H.loadSdkIframeEmbedTestPage({
+      dashboardId: ORDERS_DASHBOARD_ID,
+      onVisitPage: () => stubWindowOpenForSamlPopup(),
+    });
+
+    assertDashboardLoaded(frame);
+  });
+
+  it("shows an error if the SAML login results in an invalid user", () => {
+    mockAuthSsoEndpointForSamlAuthProvider();
+    H.prepareSdkIframeEmbedTest({ enabledAuthMethods: [] });
+    cy.signOut();
+
+    const frame = H.loadSdkIframeEmbedTestPage({
+      dashboardId: ORDERS_DASHBOARD_ID,
+      onVisitPage: () => stubWindowOpenForSamlPopup({ isUserValid: false }),
+    });
+
+    frame.within(() => {
+      cy.findByTestId("sdk-error-container")
+        .should("be.visible")
+        .and(
+          "contain",
+          "Failed to fetch the user, the session might be invalid.",
+        );
+    });
+  });
+
+  it("shows an error if we are using an API key in production", () => {
+    H.prepareSdkIframeEmbedTest({ enabledAuthMethods: ["api-key"] });
+    cy.signOut();
+
+    cy.log("restore the current page's domain");
+    cy.visit("http://localhost:4000");
```

Comment:
can you explain why do we need this line here? I'm not totally sure what it does, but removing it would fail the test.